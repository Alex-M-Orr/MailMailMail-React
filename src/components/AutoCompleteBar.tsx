
import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { RootStore } from '../reducers';
import { getUsers, otherUser } from '../actions/actions';
import { useHistory } from "react-router";
import { UserInfo } from '../states/states';

interface itemType {
    id: number
    name: string,

}


export const AutoCompleteBar: React.FC = () => {

    const history = useHistory();
    const [selected, setSelected] = useState(-1);

    const dispatch = useDispatch();

    useEffect(() => {
        if (selected !== -1) {
            dispatch(otherUser(selected));
            history.push("/other");
        }
    }, [dispatch, selected, history])

    useEffect(() => {
        dispatch(getUsers());
    }, [dispatch])

    const usersState = useSelector((state: RootStore) => state.users)
    let items: itemType[] = [];
    const funct = () => {

        function isRealUser(element:UserInfo, index:number, array:any) {
            return (element.id !== 23 && element.id !== 15);
        } 
        const filtUsers: UserInfo[] = usersState.users.filter(isRealUser);

        filtUsers.forEach(function (user) {
            let x: itemType = {
                id: user.id,
                name: `${user.firstName} ${user.lastName}`
            }
            items.push(x);
        })
        return items;
    };

    //maybe move funct into a useEffect with a dependency on usersState. 
    funct();
    const [suggestions, setSuggestions] = useState<itemType[]>([])
    const [text, setText] = useState<string>("");

    const onTextChanged = (e: any) => {
        const value = e.target.value;
        let suggs: itemType[] = [];
        if (value.length > 0) {
            const regex = new RegExp(`^${value}`, 'i');
            suggs = items;
            suggs.sort((a, b) => {
                if (a.name > b.name) {
                    return 1;
                } else if (a.name < b.name) {
                    return -1;
                }
                return 0;
            })
            suggs = suggs.filter(v => regex.test(v.name));
        }
        setSuggestions(suggs);
        setText(value);
    }

    function suggestionSelected(value: itemType): any {
        setText(value.name);
        setSuggestions([]);
        setSelected(value.id);

    }


    const renderSuggestions = () => {
        if (suggestions.length === 0) {
            return null;
        }
        return (
            <ul>
                {suggestions.map((item: itemType) => <li onClick={() => suggestionSelected(item)} key={item.id}>{item.name}</li>)}
            </ul>
        )
    }

    return (
        <>
            <div className="AutoCompleteText">
                <input value={text} onChange={onTextChanged} type="text" placeholder="Search Users..." />
                {renderSuggestions()}
            </div>
        </>
    )



}