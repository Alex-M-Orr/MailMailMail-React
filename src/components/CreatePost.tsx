import React from "react";
import {
  Card,
  Container,
  Row,
  Col,
  Form,
  Button,
  Image,
} from "react-bootstrap";
import { useSelector } from "react-redux";
import configData from "../config.json";
import { RootStore } from "../reducers";

const axios = require("axios");

export const CreatePost: React.FC = () => {
  const user = useSelector((state: RootStore) => state.login);

  const submitPost = async (event: any) => {
    event.preventDefault();
    const form = event.currentTarget;
    console.log(form);
    const newPost = {
      content: form[0].value,
      photo: form[1].value,
      authorId: user.id,
      datecreated: null,
    };
    try {
      const a = await axios.post(
        configData.SERVER_URL + "/postSave.app",
        newPost
      );
      console.log(a.data);
    } catch (error) {
      console.log(error);
    }
  };

  return (
    <>
      <Card>
        <Card.Body>
          <Container>
            <Row>
              <Col lg={2}>
                <Image
                  className="mb-2 mr-2"
                  src={configData.S3_URL +user.photo}
                  width="150"
                  roundedCircle
                />
              </Col>
              <Col>
                <h5>
                  {user.firstName} {user.lastName}
                </h5>
                <Form onSubmit={submitPost}>
                  <Container>
                    <Row>
                      <Col>
                        <Form.Group controlId="exampleForm.ControlTextarea1">
                          <Form.Control
                            as="textarea"
                            rows={3}
                            placeholder="Write post here..."
                          />
                        </Form.Group>
                      </Col>
                    </Row>
                    <Row>
                      <Form.Group className="w-100">
                        <Form.File className="mb-3" id="formcheck-api-regular">
                          <Form.File.Input />
                        </Form.File>
                        <Button type="submit" className="blue float-right">
                          Create Post
                        </Button>
                      </Form.Group>
                    </Row>
                  </Container>
                </Form>
              </Col>
            </Row>
          </Container>
        </Card.Body>
      </Card>
    </>
  );
};
