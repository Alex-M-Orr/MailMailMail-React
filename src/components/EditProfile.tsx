import React, { useState } from "react";
import { Button, Form, Modal, Image} from "react-bootstrap";

export const EditProfile: React.FC = (props: any) => {
  const [show, setShow] = useState(false);
  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);



//   const [password, setPassword] = useState(props.password);
//   const [fname, setFname] = useState(props.fname);
//   const [lname, setLname] = useState(props.lname);
//   const [photo, setPhoto] = useState(props.photo);
const [password, setPassword] = useState("wasspord");
const [fname, setFname] = useState("Enoch");
const [lname, setLname] = useState("Cho");
const [photo, setPhoto] = useState("/images/defaultImage.svg");
const [changed, setChanged] = useState(true);


  const submit = (event:any) => {
      setShow(false);
      event.preventDefault();
      const form = event.currentTarget;
      if(form[0].value.trim()){
          setPassword(form[0].value.trim());
      }
      if(form[1].value.trim()){
          setFname(form[1].value.trim());
      }
      if(form[2].value.trim()){
          setLname(form[2].value.trim());
      }
      if(form[3].value.trim()){
          setPhoto(form[3].value);
      }
      setChanged(!changed);
  }
  


  return (
    <>
      <Button size="lg" className="blue" onClick={handleShow}>
        Edit M3
      </Button>

      <Modal
        show={show}
        onHide={handleClose}
        size="lg"
        aria-labelledby="contained-modal-title-vcenter"
        centered
      >
        <Modal.Header closeButton className="blue">
          <Modal.Title>Edit Your Profile</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <Form onSubmit= {submit}>
            <Form.Group controlId="formPassword">
              <Form.Label>Password: {password}</Form.Label>
              <Form.Control type="password" placeholder="Enter new password"/>
            </Form.Group>
            <Form.Group controlId="formFirstName">
              <Form.Label>First Name: {fname}</Form.Label>
              <Form.Control type="text" placeholder="Enter new first name"/>
            </Form.Group>
            <Form.Group controlId="formLastName">
              <Form.Label>Last Name: {lname}</Form.Label>
              <Form.Control type="text" placeholder="Enter new last name"/>
            </Form.Group>
            <Form.Group controlId="formPhoto">
              <Form.Label>Profile Picture:</Form.Label>
              <Image
                        src= {photo}
                        width="150"
                        rounded
                      />
              <Form.File id="newPic"/>
            </Form.Group>
            <Button className="blue" type="submit">
            Save
          </Button>
          </Form>
        </Modal.Body>
      </Modal>
    </>
  );
};
