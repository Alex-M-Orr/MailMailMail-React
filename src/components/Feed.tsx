import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getFeed } from '../actions/actions';
import { RootStore } from '../reducers';
import { Post } from "./Post";
import {PostInfo, LikeInfo} from '../states/states';


//Id of the profile's user. if it's 0, then it's the home page feed so all posts should be shown. 
interface IProps {
    id?: number
}


export const Feed: React.FC<IProps> = (props:IProps) =>{
 
    const dispatch = useDispatch();
    const feedState = useSelector((state:RootStore) => state.feed)
    //this should run once to load the posts. 
    useEffect(() => {
        props.id?dispatch(getFeed(0)):dispatch(getFeed(0));
    }, [dispatch, props.id])

    

    return (
        <>
        {feedState.postsAndLikes && (
            <div className="mb-5">
                {feedState.postsAndLikes.map((e:[PostInfo, LikeInfo[]], i:number) => {
                    return (<Post id={i} key={i}/>)
                })}
            </div>

        )}
        </>
    )
}