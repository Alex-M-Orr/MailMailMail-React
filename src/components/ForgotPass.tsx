import React, { useState } from "react";
import { Button, Form, Modal } from "react-bootstrap";

const axios = require("axios");

export const ForgotPass: React.FC = () => {
  const [show, setShow] = useState(false);
  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);

  const resetPass = (event: any) => {
    event.preventDefault();
    try {
      const url =
        "http://18.191.119.230:8081/Project2-1.0.0/forgotPass?email=" +
        event.currentTarget[0].value;
      axios.post(url);
      alert("Password reset instruction is sent to your email!");
    } catch (error) {
      alert("Account does not exist!");
    }
  };

  return (
    <>
      <Button variant="outline-primary" onClick={handleShow}>
        Forgot password?
      </Button>
      <Modal
        show={show}
        onHide={handleClose}
        size="lg"
        aria-labelledby="contained-modal-title-vcenter"
        centered
      >
        <Modal.Header className="blue" closeButton>
          <Modal.Title>Reset Password</Modal.Title>
        </Modal.Header>

        <Modal.Body>
          <Form onSubmit={resetPass}>
            <Form.Group controlId="exampleForm.ControlInput1">
              <Form.Label>Email address</Form.Label>
              <Form.Control
                type="email"
                placeholder="Enter your account's email..."
              />
            </Form.Group>
            <Button className="blue" type="submit">
              Submit
            </Button>
          </Form>
        </Modal.Body>
      </Modal>
    </>
  );
};
