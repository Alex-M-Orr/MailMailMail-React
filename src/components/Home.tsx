import React from "react";
import {
  Button,
  Card,
  Col,
  Container,
  Form,
  Image,
  Row,
} from "react-bootstrap";
import { NavBar } from "./NavBar";
import { Post } from "./Post";

export const Home: React.FC = () => {
  return (
    <>
      <NavBar />
      <Container className="mt-5">
        <Row>
          <Col>
            <Card>
              <Card.Body>
                <Container>
                  <Row>
                    <Col lg={2}>
                      <Image
                        className="mb-2 mr-2"
                        src="/defaultImage.svg"
                        width="150"
                        rounded
                      />
                    </Col>
                    <Col>
                      <h5>Author</h5>
                      <Form>
                        <Form.Group controlId="exampleForm.ControlTextarea1">
                          <Form.Control
                            as="textarea"
                            rows={3}
                            placeholder="Write post here..."
                          />
                        </Form.Group>
                        <Form.Group>
                          <Container>
                            <Row>
                              <Col>
                                <Form.File id="formcheck-api-regular">
                                  <Form.File.Input />
                                </Form.File>
                              </Col>
                              <Col>
                                <Button className="blue float-right">Create Post</Button>
                              </Col>
                            </Row>
                          </Container>
                        </Form.Group>
                      </Form>
                    </Col>
                  </Row>
                </Container>
              </Card.Body>
            </Card>
          </Col>
        </Row>
        <Row>
            <Col>
                {/* Post Here */}
                <Post/>
                <Post/>
            </Col>
        </Row>
      </Container>
    </>
  );
};
