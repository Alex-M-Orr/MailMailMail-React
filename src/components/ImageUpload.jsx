import React from "react";
import { Button, Form, Modal } from "react-bootstrap";
import S3 from "react-aws-s3";
import configData from "../config.json";

const axios = require("axios");
const url =
  configData.SERVER_URL + "/userByEmail.app?email=sneakykey@email.com";
export const ImageUpload = (props) => {
  // ReactS3Client.uploadFile(file, newFileName).then(data => console.log(data).catch(err => console.error(err)));

  const upload = async (e) => {
    let updateUser = JSON.parse(sessionStorage.getItem('user'));
    try {
      const response = await axios.get(url);
      const notakey = await response.data;

      const config = {
        bucketName: "mailmailmail-images",
        dirName: "profileImages",
        region: "us-east-2",
        accessKeyId: notakey.firstName,
        secretAccessKey: notakey.lastName,
      };
      
      const ReactS3Client = new S3(config);
      const newFileName = JSON.parse(sessionStorage.getItem('user')).id;
     const data = await ReactS3Client.uploadFile(e.target.files[0], newFileName);
     console.log(data);
     updateUser.photo = data.key;
     await axios.put(configData.SERVER_URL + "/updateUser.app", updateUser);

    } catch (error) {
      console.log("====================================");
      console.log(error);
      console.log("====================================");
    }
  };

  return (
    <>
      <Form.File id="newPic" onChange={upload} />
    </>
  );
};
