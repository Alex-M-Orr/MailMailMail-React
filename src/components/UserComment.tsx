import React from "react";
import { Card, Image } from "react-bootstrap";

export const UserComment: React.FC = () => {
  return (
    <>
      <Card>
        <Card.Body>
          <Image src="/defaultImage.svg" width="50" height="50" rounded />
          <h6 className="m-2 d-inline">Author</h6> Nice post
        </Card.Body>
      </Card>
    </>
  );
};
