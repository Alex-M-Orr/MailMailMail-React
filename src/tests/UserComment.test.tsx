import React from "react";
import { shallow } from "enzyme";
import { UserComment } from "../components/UserComment";


test("renders the UserComment component", () => {
    const component = shallow(<UserComment/>);
    expect(component).toMatchSnapshot();
});