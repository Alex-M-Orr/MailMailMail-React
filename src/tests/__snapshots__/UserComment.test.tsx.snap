// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UserComment component 1`] = `
<Fragment>
  <Card
    body={false}
  >
    <CardBody>
      <Image
        fluid={false}
        height="50"
        rounded={true}
        roundedCircle={false}
        src="/defaultImage.svg"
        thumbnail={false}
        width="50"
      />
      <h6
        className="m-2 d-inline"
      >
        Author
      </h6>
       Nice post
    </CardBody>
  </Card>
</Fragment>
`;
